Microsoft.Azure.Cosmos/src/Diagnostics/CosmosDiagnostics.cs

```diff
@@ -27,6 +27,33 @@ public virtual TimeSpan GetClientElapsedTime()
         /// Gets the string field <see cref="CosmosDiagnostics"/> instance in the Azure CosmosDB database service.
         /// </summary>
         /// <returns>The string field <see cref="CosmosDiagnostics"/> instance in the Azure CosmosDB database service.</returns>
+        /// <remarks>
+        /// <see cref="CosmosDiagnostics"/> implements lazy materialization and is only materialized when <see cref="CosmosDiagnostics.ToString"/> is called.
+        /// </remarks>
+        /// <example>
+        /// Do not eagerly materialize the diagnostics until the moment of consumption to avoid unnecessary allocations, let the ToString be called only when needed.
+        /// You can capture diagnostics conditionally, based on latency or errors:
+        /// <code language="c#">
+        /// <![CDATA[
+        /// try
+        /// {
+        ///     ItemResponse<Book> response = await container.CreateItemAsync<Book>(item: testItem);
+        ///     if (response.Diagnostics.GetClientElapsedTime() > ConfigurableSlowRequestTimeSpan)
+        ///     {
+        ///         // Log the diagnostics and add any additional info necessary to correlate to other logs 
+        ///         log.LogInformation("Operation took longer than expected, Diagnostics: {Diagnostics}", response.Diagnostics);
```

Comment:
should this be logger to match the catch? It shouldn't be passed in as an arg.
```suggestion
        ///         logger.LogInformation("Operation took longer than expected, Diagnostics: {Diagnostics}");
```